### DISCUSSION TITLE
Scripting kitty with sessions: how to avoid a window to close after sending a SIGTERM signal (CTRL-C)

### DISCUSSION THREAD

**Discussion by kamatheuska** [kovidgoyal/kitty#5926] [Q&A]
I am creating scripts that allow me to have sessions with kitty.

This is what I have done until now:

```zsh
# ~/config/kitty/session.conf
 layout tall

cd /path/to/folder
launch zsh

# Open another window inside this new os window
cd ./some/relative/path
launch zsh -c 'pnpm dev; zsh -l -i;'
```

Then, to start the session, I do

```zsh
# start the session (not sure how to do that either, but this works for now)
kitty --session session.conf
```

One issue is that the created window does not stay open if the process is terminated. The window **briefly flashes and is closed immediately**.

How would you script that kind of behavior?

Also, I could not find more lengthy examples on how to write sessions, or script kitty, do we have something like that on some README?

**Comment by kovidgoyal** [comment]
kitty --detach

**Reply by kamatheuska** [reply]
@kovidgoyal Thanks for the quick answer! I guess what you mean is to start my session with

```zsh
kitty --detach --session session.conf
```

That does not prevent a window to close after CTRL-C, that is started as

```zsh
launch pnpm dev
```

I am missing something?

**Comment by kovidgoyal** [comment]
On Mon, Jan 23, 2023 at 11:24:39PM -0800, Nicolas Ramirez wrote:
> @kovidgoyal Thanks for the quick answer! I guess what you mean is to start my session with
> 
> ```zsh
> kitty --detach --session session.conf
> ```
> 
> That does not prevent a window to close that is started as
> 
> ```zsh
> launch pnpm dev
> ```
> 
> I am missing something?

launch sh -c "pnpn dev; bash"
